python/src/aac/plugins/gen_plant_uml/gen_plant_uml_impl.py

```diff
@@ -19,9 +20,10 @@
 COMPONENT_STRING = "component"
 OBJECT_STRING = "object"
 SEQUENCE_STRING = "sequence"
+INVALID_FILE_NAME_CHARACTERS = ".!@#$%^&*();,\\/?[]{}`~|'"
 
 
-def puml_component(architecture_file: str, output_directory: str = None) -> PluginExecutionResult:
+def puml_component(architecture_file: str, output_directory: Union[str, None] = None) -> PluginExecutionResult:
```

Comment:
You can use Optional here instead.
```suggestion
def puml_component(architecture_file: str, output_directory: Optional[str] = None) -> PluginExecutionResult:
```